modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -121,6 +121,12 @@
 #define mingw_getsp(...) 0
 #define __builtin_frame_address(...) 0
 
+static bool isLittleEndian()
```

Comment:
There is an endianness check function already by the way.
https://github.com/opencv/opencv/blob/459a60927135d9aaa3d4a8bc2b46e36dbba6f7f2/modules/imgcodecs/src/utils.hpp#L138-L145

The nice thing about this one is that endianness can be inferred at compile-time and so the compiler can optimize away the two if-branches on big endian platforms.